core/mock/mock.go

```diff
@@ -54,6 +54,7 @@ func NewMockNode() (*core.IpfsNode, error) {
 	nd.Peerstore = peer.NewPeerstore()
 	nd.Peerstore.AddPrivKey(p, ident.PrivateKey())
 	nd.Peerstore.AddPubKey(p, ident.PublicKey())
+	nd.Identity = p
```

Comment:
So this means any tests using `coremock.NewMockNode()` loads private key, while the tests that use `core.NewIPFSNode` (e.g. https://github.com/ipfs/go-ipfs/blob/master/core/coreunix/add_test.go#L28) don't.